File: is-subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        if(s.empty())return true;
        int index = 0;
        for(int i=0;i<t.size();i++)
        {
            if(t[i] == s[index])index++;
            if(index == s.size())return true;
        }
        return false;
    }
};

File: June LeetCoding Challenge/Is Subsequence.cs
public class Solution {
    public bool IsSubsequence(string s, string t) {
        int idx = 0;
        foreach(char c in t.ToCharArray())
        {
            if(idx == s.Length)
                return true;
            if(c == s[idx])
                idx++;
        }
        return idx == s.Length;
    }
}
